backend/corpora/common/entities/collection.py

```diff
@@ -344,6 +351,33 @@ def update(self, links: list = None, keep_links=False, **kwargs) -> None:
 
         super().update(**kwargs)
 
+    def update_curation(self, links: list = None, keep_links=False, **kwargs) -> None:
+        """
+        Update an existing collection to match provided the parameters. The specified columns are replaced. The DOI is
+        handled as a link for legacy reasons (for consistency with Corpora API), but independently of the other links.
+        :param links: links to create and connect to the collection. If present, the existing attached entries will
+         be removed and replaced with new entries.
+        :param keep_links: boolean - whether or not links need to be preserved. Links are preserved if True.
+        :param kwargs: Any other fields in the dataset that will be replaced.
+        """
+        links = links if links else []
+
+        if not keep_links:
+            for link in self.links:
+                if link.link_type != ProjectLinkType.DOI:
+                    self.session.delete(link)
+
+        for link in links:
+            if link["link_type"] == ProjectLinkType.DOI.name:
+                for link in self.links:
+                    if link.link_type == ProjectLinkType.DOI:
+                        self.session.delete(link)
```

Comment:
Yes cc @brianraymor for confirmation

Reply:
I am pretty sure the only reason we support multiple DOIs at the moment is because we don't have a mechanism to prevent duplication of link types. The current code just uses the first DOI when populating the Collection with publisher metadata if I recall correctly, anyways.

Reply:
@ebezzi  implemented these changes, but the requirement was that there can only be one DOI per collection per https://github.com/chanzuckerberg/single-cell/issues/214:

> only allow one DOI link to be added to the collection

Reply:
It sounds like we can avoid these checks for duplicate DOIs

Reply:
Yep, correct. I've just confirmed that there are no (legacy) instances of multiple DOIs for a single Collection in prod, and the code (both Corpora and Curation) no longer allows multiple DOIs per Collection. So we're good. 👍